tools/plugins/com.liferay.ide.ui/src/com/liferay/ide/ui/LiferayUIPlugin.java

```diff
@@ -339,6 +341,28 @@ public void run() {
 			});
 	}
 
+	private void _enableGradleAutoSync() {
+		IEclipsePreferences gradlePrefs = InstanceScope.INSTANCE.getNode("org.eclipse.buildship.core");
+
+		String autoSync = "auto.sync";
+
+		try {
+			String[] gradlePrefsKeys = gradlePrefs.keys();
+
+			Stream<String> gradlePrefsKeysSteam = Stream.of(gradlePrefsKeys);
```

Comment:
This should be done in the product_customization.ini file here:  https://github.com/liferay/liferay-ide/blob/master/enterprise/plugins/com.liferay.ide.studio.ui/product_customization.ini

Reply:
Hey ,greg. 
    I tried to use what you said  to implement, but it couldn't be done. The reason:
buildship not use `UIPlugin.getInstance().getPreferenceStore()` to receive client's inputs.
then i tried to change buildship source code to verify it. Is my explanation correct?  You can reference this class [GradleWorkbenchPreferencePage ](https://github.com/eclipse/buildship/blob/master/org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/preferences/GradleWorkbenchPreferencePage.java#L75).